### PR TITLE
Magnetometer can use the screen local coordinates

### PR DESCRIPTION
**Pull request by alexshalamov**
Introduce MagnetometerSensorOptions that can control whether screen
or device coordinate system is used.

This patch is a part of fix for https://github.com/w3c/sensors/issues/257

***
<a href="https://pr-preview.s3.amazonaws.com/alexshalamov/magnetometer/pull/39.html" title="Last updated on Feb 8, 2018, 12:44 PM GMT (66905ee)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/magnetometer/39/7dbf5d6...alexshalamov:66905ee.html" title="Last updated on Feb 8, 2018, 12:44 PM GMT (66905ee)">Diff</a>

### CODE REVIEW

**Comment by alexshalamov** [comment]
@anssiko @pozdnyakov @riju ptal.

**Comment by anssiko** [comment]
See https://github.com/w3c/accelerometer/pull/34#pullrequestreview-94715890

File: index.bs [hunk 1]
```diff
@@ -28,9 +28,16 @@ urlPrefix: https://w3c.github.io/sensors; spec: GENERIC-SENSOR
     text: sensor
     text: latest reading
     text: default sensor
-    text: construct a sensor object; url: construct-sensor-object
     text: sensor type
     text: mitigation strategies; url: mitigation-strategies
+    text: local coordinate system
+urlPrefix: https://w3c.github.io/accelerometer; spec: ACCELEROMETER
```
Comment (riju) [inline index.bs]: Let's put a note in Accelerometer, why we are using SpatialSensor in Accelerometer spec.

File: index.bs [hunk 2]
```diff
@@ -161,13 +167,16 @@ For uncalibrated magnetometer, the [=latest reading=] includes three [=map/entri
...
 
-Note: The <dfn>local coordinate system</dfn> of a mobile device is usually defined relative to
-the device's screen when the device in its default orientation (see figure below).
+<img src="images/magnetometer_coordinate_system.svg" onerror="this.src='images/magnetometer_coordinate_system.png'" style="display: block;margin: auto;" alt="Magnetometer coordinate system.">
 
-<img src="images/magnetometer_coordinate_system.png" srcset="images/magnetometer_coordinate_system.svg" alt="Magnetometer coordinate system.">
+Reference Frame {#reference-frame}
+----------------
 
+The [=local coordinate system=] represents the reference frame for the [=sensor readings=].
```
Comment (pozdnyakov) [inline index.bs]: maybe for "for Magnetometer readings" ?

File: index.bs [hunk 3]
```diff
@@ -176,33 +185,24 @@ The Magnetometer Interface {#magnetometer-interface}
...
-    readonly attribute double? zBias;
+  dictionary MagnetometerSensorOptions :  SensorOptions  {
+    LocalCoordinateSystem referenceFrame = "device";
   };
 </pre>
 
-
-To <dfn>Construct a Magnetometer Object</dfn>, or to <dfn>Construct an UncalibratedMagnetometer Object</dfn> the user agent must invoke the <a>construct a Sensor object</a> abstract operation.
-
+To <dfn>Construct a Magnetometer Object</dfn> the user agent must invoke the
```
Comment (pozdnyakov) [inline index.bs]: do we still need \<dfn>Construct a Magnetometer Object\</dfn> here ?

maybe just "To construct a {{Magnetometer}} object .."

File: index.bs [hunk 4]
```diff
@@ -176,33 +185,24 @@ The Magnetometer Interface {#magnetometer-interface}
...
+[=construct magnetometer sensor object=] abstract operation for {{Magnetometer}}
```
Comment (pozdnyakov) [inline index.bs]: [=construct a magnetometer object=]

for the {{Magnetometer}} interface

File: index.bs (hunk 2 above)
Comment (anssiko) [inline index.bs]: You want to agree on a consistent language with https://github.com/w3c/gyroscope/pull/29/files#diff-ec9cfa5f3f35ec1f84feb2e59686c34dR107

Comment (alexshalamov) [reply index.bs]: done

Comment (alexshalamov) [reply index.bs]: done

File: index.bs (hunk 3 above)
Comment (alexshalamov) [reply index.bs]: done

File: index.bs (hunk 4 above)
Comment (alexshalamov) [reply index.bs]: done

**Pull request merged**